lib/bridge/MatrixHandler.js

```diff
@@ -770,6 +769,68 @@ MatrixHandler.prototype._onMessage = Promise.coroutine(function*(req, event) {
     yield Promise.all(promises);
 });
 
+MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
+    function*(req, ircRoom, ircUser, ircAction, event) {
+    let self = this;
+
+    if (event.content.msgtype !== "m.text" ||
+        !(ircUser.unsafeClient && ircUser.unsafeClient.wouldSend)) {
+        yield self.ircBridge.sendIrcAction(ircRoom, ircUser, ircAction);
+    }
+    else {
+        let text = event.content.body;
+        let wouldSend = ircUser.unsafeClient.wouldSend(ircRoom.channel, text);
+        let lineLimit = 2;
+        if (wouldSend.length > lineLimit) {
+            // Message body too long - send a file to HS instead and send link to IRC
+
+            try {
+                // Upload as a file and get URI
+
+                let fileName = Date().replace(/\s/g, '_') + ".txt";
+                let response = yield self.ircBridge.sendMatrixPlaintextFile(fileName, text);
+                let result = JSON.parse(response);
+                event.content.url = result.content_uri;
+
+                // Send the link to IRC
+                event.content.msgtype = "m.file";
+                event.content.body = fileName;
+
+                let bigFileIrvAction = IrcAction.fromMatrixAction(
+                    MatrixAction.fromEvent(
+                        self.ircBridge.getAppServiceBridge().getClientFactory().getClientAs(), event
+                    )
+                );
+
+                yield self.ircBridge.sendIrcAction(ircRoom, ircUser, bigFileIrvAction);
+            }
+            catch (err) {
+                req.log.error("Failed to upload file, will truncate and send instead", err);
+
+                event.content = {
+                    url : undefined,
+                    msgtype : "m.text",
+                    body : wouldSend.splice(0, lineLimit).join('\n') +
+                     "\n(Could not convert large text (truncated above) to file: \"" +
+                     err.message +
+                     "\")"
+                };
+
+                let trucatedIrcAction = IrcAction.fromMatrixAction(
+                    MatrixAction.fromEvent(
+                        self.ircBridge.getAppServiceBridge().getClientFactory().getClientAs(), event
+                    )
+                );
+
+                yield self.ircBridge.sendIrcAction(ircRoom, ircUser, trucatedIrcAction);
+            }
+        }
+        else {
+            yield self.ircBridge.sendIrcAction(ircRoom, ircUser, ircAction);
+        }
```

Comment:
Invert your if statement so you can put this first. There are many lines between when you did that check and here, making it non-obvious when this will be executed. E.g:
```javascript
if (wouldSend.length <= lineLimit) {
  yield self.ircBridge.sendIrcAction(ircRoom, ircUser, ircAction);
  return;
}
// text is too long and needs to be uploaded
// <upload code>
```